### PR TITLE
homework 4 

### PR DESCRIPTION
**Pull request by peterrowland**
now with ex08_05

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,19 +11,48 @@
 #         - then ends the program
 ################################################################################
 # Imports
-
+import random as random
 
 # Body
+# generates random interger 1 - 25
+secret_number = (random.randint(1,25))
```
Comment (danielsgriffin) [inline D04_ex00.py]: This would really benefit from being broken into functions.

File: D04_ex00.py [hunk 2]
```diff
@@ -11,19 +11,48 @@
...
+
+    #tell user if correct, high, low
+    if guess == secret_number:              # correct case, exits program
+        print("You guessed it! Goodbye.")
+        break
+    elif guess <= secret_number:            # guess low
+        count_guess -= 1
+        print("Higher.\nYou have " + str(count_guess) + " guesses left." )
+        guess = ''
+    elif guess >= secret_number:            # guess high
```
Comment (danielsgriffin) [inline D04_ex00.py]: Should this be elif?

File: HW04_ch07_ex02.py [hunk 3]
```diff
@@ -21,15 +21,24 @@
...
 ###############################################################################
 # Imports
-
+import math
 
 # Body
-
+def eval_loop ():
+    #init input
+    eval_input = ''
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "continue until the user enters 'done', and then return the value of the last expression it evaluated"

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -2,3 +2,33 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]:
```
Book Examples:
  X  'A'     by 3   => got: 'd'     expected: 'D'
  X  'Z'     by 1   => got: 'a'     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
 OK  'melon' by -10 => got: 'cubed' expected: 'cubed'
  X  'IBM'   by -1  => got: 'hal'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
  X  'xyz'   by 1   => got: 'y`a'   expected: 'yza'
  X  'ABC'   by 1   => got: 'bcd'   expected: 'BCD'
  X  'XYZ'   by 1   => got: 'y`a'   expected: 'YZA'
  X  'Az'    by 1   => got: 'ba'    expected: 'Ba'
  X  'Aa'    by 26  => got: 'aa'    expected: 'Aa'
  X  'abc'   by -1  => got: '`ab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: '`ab'   expected: 'ZAB'
  X  'XYZ'   by -1  => got: 'wxy'   expected: 'WXY'
  X  'Az'    by -1  => got: '`y'    expected: 'Zy'
  X  'Aa'    by -26 => got: 'aa'    expected: 'Aa'
```

**Comment by danielsgriffin** [comment]
Good job. Ensure you have thorough testing of your functions.

File: HW04_string2.py [hunk 5]
```diff
@@ -30,9 +37,11 @@ def verbing(s):
 # So 'This dinner is not that bad!' yields:
 # This dinner is good!
 def not_bad(s):
-    # +++your code here+++
-    return
+    # find 'not'
+    if (s.find('not') and s.find('bad')) and (s.find('not') < s.find('bad')):
+        s = s.split('not')[0] + 'good' + s.split('bad')[1]
```
Comment (harmanssingh) [inline HW04_string2.py]: another way:  s.split("not")[0]+"good" s[s.find("bad")+3:]

File: D04_ex00.py [hunk 6]
```diff
@@ -11,19 +11,48 @@
...
+                print('Please guess a number between 1 - 25')
+                guess = ''
+        except:
+            print('Please enter an integer.')
+
+    #tell user if correct, high, low
+    if guess == secret_number:              # correct case, exits program
+        print("You guessed it! Goodbye.")
+        break
+    elif guess <= secret_number:            # guess low
```
Comment (harmanssingh) [inline D04_ex00.py]: if guess = secret number in guess<=secret number, you already guessed it right. I feel the condition should be guess<secret number.

File: HW04_ch08_ex05.py [hunk 7]
```diff
@@ -2,3 +2,33 @@
...
 # function.
+
+def rotate_word(word,n):
+    word = word.lower()
```
Comment (harmanssingh) [inline HW04_ch08_ex05.py]: You don't need a lower here. Also if the letter is Capital, the offset changes.

**Pull request closed without merging**